Presto Oracle connector

sajjoseph opened the pull request:
Implements Oracle connector using Oracle JDBC driver. Tested against Oracle 12c even though it should work with older versions as well.

electrum commented:
Thanks for contributing this!

Let's start with the obvious issue that the Oracle JDBC driver isn't available under an OSS-friendly license, thus it's not available in Maven central and we can't redistribute it with Presto. This creates some problems for us to solve:

1. How to build? We need to build Presto without the Oracle driver (e.g., Travis CI needs to pass). Fortunately, the dependency is not required for compilation, so an easy solution is to simply remove the driver dependency from the POM.

2. How do end users use it? We can include instructions in the documentation to add the Oracle driver JAR file to the `plugin/oracle` directory on the Presto coordinator and workers. This is really bad from a user experience, but I don't see a better alternative.

3. How do we test? We need to make sure the connector doesn't break when changes are made to `presto-base-jdbc`, the SPI, library upgrades, etc. Hopefully, we can test it in Travis, similar to how we do it for other integration tests. The SQL Server test is a good one to look at. Oracle seems to have some Docker images available: https://container-registry.oracle.com/

pdanilew commented:
@sajjoseph: this connector would help a lot in my opinion, are you going to continue work on this PR?

sajjoseph commented:
Thanks for checking my contribution and the feedback.

About Problem #1, how can we avoid the dependency on Oracle Driver? In
OracleClient.java, we have the following line in line #93:
       return new DriverConnectionFactory(new OracleDriver(),
config.getConnectionUrl(), connectionProperties);
I wonder, how we can resolve the name OracleDriver if we don't include the
Oracle driver dependency. Please share your thoughts.

About #2, I will update the README with instructions for end users to
deploy the Oracle Driver file separately.
For #3, I am looking into how SQL Server tests are managed in
presto-product-tests module and will see if I can incorporate similar
changes for Oracle.

On Thu, Sep 27, 2018 at 10:11 AM pdanilew <notifications@github.com> wrote:

> @sajjoseph <https://github.com/sajjoseph>: this connector would help a
> lot in my opinion, are you going to continue work on this PR?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/prestodb/presto/pull/11540#issuecomment-425151736>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQhEuBzr_Z9iM2opTbCkwS9xLiPI4wpBks5ufPi4gaJpZM4WyFSL>
> .
>


-- 
Sajumon Joseph
303-378-0593
sajjoseph@gmail.com

sajjoseph commented:
#1 is fixed.

Updated the code base to remove the Oracle JDBC dependency. Travis builds are failing with presto-proxy module. Forcing another build through Travis now. Following was the reason for the previous build failure. 

[ERROR] Tests run: 8, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 25.427 s <<< FAILURE! - in TestSuite
[ERROR] testPartialCancel(com.facebook.presto.proxy.TestProxyServer)  Time elapsed: 7.071 s  <<< FAILURE!
org.testng.internal.thread.ThreadTimeoutException: Method com.facebook.presto.proxy.TestProxyServer.testPartialCancel() didn't finish within the time-out 10000
	at com.facebook.presto.proxy.TestProxyServer.testPartialCancel(TestProxyServer.java:194)

#2 
Updated the documentation in README to include instructions on copying oracle jdbc driver to the distro.

#3. 
Exploring this option.
